### ISSUE TITLE
S3440 FN: Add test cases for is expression

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6390]
### Description

FN for S3440 (RedundantConditionalAroundAssignment).

Using C#11 list pattern, the rule is not raising an issue.

### Repro steps

```csharp
      public void MyMethod(byte[] bytes)
      {
          if (bytes is [not 1, .., not 3])
          {
              bytes[0] = 1; // FN
              bytes[^1] = 3; // FN
          }
      }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by cristian-ambrosini-sonarsource** [comment]
This issue is not C#11 related. The rule does not support is expression yet, another issue will be opened to fix that and it will be linked below.
